test/conv_common.hpp

```diff
@@ -1712,6 +1713,27 @@ struct conv_driver : test_driver
         exit(-1); // NOLINT (concurrency-mt-unsafe)
     }
 
+    ///\brief We only support fp16x4 and fp16x8 vector type
+    void find_vector_length_from_conv_mode()
+    {
+        if(conv_mode.compare(0, 4, "conv") == 0)
+        {
+            auto found_vec = conv_mode.find("fp16x");
+            if(found_vec != std::string::npos)
+            {
+                std::string vec_str = conv_mode.substr(found_vec + 5);
+                ///\todo try catch()
+                vector_length = std::stoi(vec_str);
+                if(vector_length != 4 && vector_length != 8)
+                    MIOPEN_THROW("Unsupported vector length");
+            }
```

Comment:
Vector length should be its own argument to the test and not attached to the convolution mode.